lib/push.js

```diff
@@ -484,11 +484,9 @@ module.exports = function (log, db, config) {
               // the push settings need to be reset.
               // the clients will check this and re-register push endpoints
               if (err.statusCode === 404 || err.statusCode === 410 || keyWasInvalid) {
-                // reset device push configuration
+                // set the push endpoint expired flag
                 // Warning: this method is called without any session tokens or auth validation.
-                device.pushCallback = ''
-                device.pushPublicKey = ''
-                device.pushAuthKey = ''
+                device.pushEndpointExpired = true
```

Comment:
One minor addition to this logic: of the push endpoint is marked as expired, we should skip trying to send more push messages to that device, because we know they'll fail.